hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -142,13 +142,19 @@ public void accept(MessageSendingResult result) {
                 handleFailedSending(message, result);
                 if (!isTtlExceeded(message) && shouldRetrySending(result)) {
                     retrySingleThreadExecutor.schedule(() -> retrySending(result),
-                            subscription.getSubscriptionPolicy().getMessageBackoff(), TimeUnit.MILLISECONDS);
+                            extractRetryDelay(result), TimeUnit.MILLISECONDS);
```

Comment:
Remember that if you schedule retrySending then actual attempt will happen because TTL is checked when processing response... We don't have any logic that would clear jobs from retry executor...